app/src/main/java/org/mozilla/fenix/home/sessioncontrol/SessionControlController.kt

```diff
@@ -510,16 +510,18 @@ class DefaultSessionControlController(
     }
 
     override fun handleShowWallpapersOnboardingDialog(state: WallpaperState): Boolean {
-        if (state.availableWallpapers.all { it.thumbnailFileState == Wallpaper.ImageFileState.Downloaded } &&
-            state.availableWallpapers.size >= WallpaperOnboardingDialogFragment.THUMBNAILS_COUNT
-        ) {
-            navController.nav(
-                R.id.homeFragment,
-                HomeFragmentDirections.actionGlobalWallpaperOnboardingDialog(),
-            )
-            return true
+        return state.availableWallpapers.filter { wallpaper ->
+            wallpaper.thumbnailFileState == Wallpaper.ImageFileState.Downloaded
+        }.size.let { downloadedCount ->
+            downloadedCount >= WallpaperOnboardingDialogFragment.THUMBNAILS_COUNT
+        }.also { isThumbnailsDownloaded ->
+            if (isThumbnailsDownloaded) {
```

Comment:
```suggestion
            if (downloadedCount >= WallpaperOnboardingDialogFragment.THUMBNAILS_COUNT) {
```
I don't feel too strongly about this, but I think we can simplify this and shave down a couple lines.

Reply:
I think we could also use a comment for `THUMBNAILS_COUNT` to indicate that this is minimum threshold to indicate that enough thumbnails have been downloaded.

Reply:
nit: maybe also rename it to make it obvious it is a minimum threshold?

Reply:
changed the param to `THUMBNAILS_SELECTION_COUNT`. It's not just a trashhold for downloads, it's the amount of items the selector is expected to have. It will take only 6 items if there are more, and we should not try to display the onboarding unless there are less then 6.

Reply:
I think this is where the comment for the parameter is more important than the proposed name change here. I don't believe the name change added any additional information or clarity for what it represents.